tests/by-util/test_split.rs

```diff
@@ -156,3 +164,63 @@ fn test_split_additional_suffix() {
     assert_eq!(glob.count(), 2);
     assert_eq!(glob.collate(), at.read(name).into_bytes());
 }
+
+#[test]
+#[cfg(unix)]
+fn test_filter() {
+    // like `test_split_default()` but run a command before writing
+    let (at, mut ucmd) = at_and_ucmd!();
+    let name = "filtered";
+    let glob = Glob::new(&at, ".", r"x[[:alpha:]][[:alpha:]]$");
+    let n_lines = 3;
+    RandomFile::new(&at, name).add_lines(n_lines);
+
+    // change all characters to 'i'
+    ucmd.args(&["--filter=sed s/./i/g > $FILE", name])
+        .succeeds();
+    // assert all characters are 'i' / no character is not 'i'
+    // (assert that command succeded)
+    assert!(
+        glob.collate().iter().find(|&&c| {
+            // is not i
+            c != ('i' as u8)
+            // is not newline
+            && c != ('\n' as u8)
+        }) == None
+    );
+}
+
+#[test]
+#[cfg(unix)]
+fn test_filter_with_env_var_set() {
+    // This test will ensure that if $FILE env var was set before running --filter, it'll stay that
+    // way
+    // implemented like `test_split_default()` but run a command before writing
+    let (at, mut ucmd) = at_and_ucmd!();
+    let name = "filtered";
+    let glob = Glob::new(&at, ".", r"x[[:alpha:]][[:alpha:]]$");
+    let n_lines = 3;
+    RandomFile::new(&at, name).add_lines(n_lines);
+
+    let env_var_value = "somevalue";
+    env::set_var("FILE", &env_var_value);
+    ucmd.args(&[format!("--filter={}", "cat > $FILE").as_str(), name])
+        .succeeds();
+    assert_eq!(glob.collate(), at.read(name).into_bytes());
+    assert!(env::var("FILE").unwrap_or("var was unset".to_owned()) == env_var_value);
+}
+
+#[test]
+#[cfg(unix)]
+fn test_filter_command_fails() {
+    let (at, mut ucmd) = at_and_ucmd!();
+    let name = "filter-will-fail";
+    RandomFile::new(&at, name).add_lines(4);
+
+    assert!(
+        !ucmd
+            .args(&["--filter=/a/path/that/totally/does/not/exist", name])
+            .fails()
```

Comment:
fails should be enough ?

fails.().success is funny :)

Reply:
then I don't `assert!` ? (there's an assertion in `fails()` anyway)